### PR TITLE
Switch to different UUID lib due to non-random generated UUIDs

### LINKED ISSUE #8077: Replace uuid library
We're using satori/go.uuid for UUIDs.

Critical bug in UUI
https://github.com/satori/go.uuid/issues/73

Evaluate other UUID libs and replace go.uuid.

### PR DESCRIPTION
**Pull request by kvch**
The previously used `github.com/satori/go-uuid` lib has a critical bug and it is no longer maintained. The community moved to an active fork of this existing lib named `github.com/gofrs/uuid`. I haven't seen other UUID libs worth mentioning apart from this.

Changes compared to the previous dependency:
* `uuid.NewV4` returns an error if the function call failed
* `uuid.Equal` is deprecated and removed. The Way is to use `==` instead.

TODO
- [x] update `github.com/elastic/go-ucfg`

Closes #8077

### CODE REVIEW

**Comment by kvch** [comment]
`github.com/elastic/go-ucfg` needs an update.

**Comment by ruflin** [comment]
Seems like `go-ucfg` uses the same library. We should do a PR there first and then update go-ucfg.

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -45,6 +45,7 @@ https://github.com/elastic/beats/compare/v6.4.0...master[Check the HEAD diff]
 - Add backoff support to x-pack monitoring outputs. {issue}7966[7966]
 - Fix a race condition with the `add_host_metadata` and the event serialization. {pull}8223[8223]
 - Enforce that data used by k8s or docker doesn't use any reference. {pull}8240[8240]
+- Switch to different UUID lib: github.com/gofrs/uuid. {pull}8485[8485]
```
Comment (ph) [inline CHANGELOG.asciidoc]: Can we insert why we switch probably adding a word about collision in the old library.

**Comment by kvch** [comment]
Updated changelog entry and rebased on top of #8469

**Comment by ph** [comment]
Make check fails on CI

File: vendor/vendor.json [hunk 2]
```diff
@@ -1094,6 +1094,12 @@
 			"revision": "ffef3ffc77bec026fefa6f76bd53d158cfa0e669",
 			"revisionTime": "2017-03-24T09:53:51Z"
 		},
+		{
+			"checksumSHA1": "BfjvAbPn5t/2KrdC5E7418OuzV8=",
+			"path": "github.com/gofrs/uuid",
```
Comment (andrewkroh) [inline vendor/vendor.json]: Can you checkout from a tag like `govendor fetch github.com/gofrs/uuid@v3.1.1`?

File: libbeat/cmd/instance/metrics_common.go [hunk 3]
```diff
@@ -35,7 +36,11 @@ func init() {
 	beatMetrics = monitoring.Default.NewRegistry("beat")
 	monitoring.NewFunc(beatMetrics, "info", reportInfo, monitoring.Report)
 
-	ephemeralID = uuid.NewV4()
+	var err error
+	ephemeralID, err = uuid.NewV4()
```
Comment (andrewkroh) [inline libbeat/cmd/instance/metrics_common.go]: What's the impact to monitoring if the ephemeral ID is Nil?

File: libbeat/outputs/elasticsearch/elasticsearch.go [hunk 4]
```diff
@@ -70,20 +70,23 @@ func newCallbacksRegistry() callbacksRegistry {
...
 	defer connectCallbackRegistry.mutex.Unlock()
 
 	// find the next unique key
 	var key uuid.UUID
 	exists := true
 	for exists {
-		key = uuid.NewV4()
+		key, err := uuid.NewV4()
+		if err != nil {
+			return uuid.UUID{}, err
```
Comment (andrewkroh) [inline libbeat/outputs/elasticsearch/elasticsearch.go]: Use `uuid.Nil` here? It's the same but slightly more clear.

File: libbeat/cmd/instance/metrics_common.go (hunk 3 above)
Comment (kvch) [reply libbeat/cmd/instance/metrics_common.go]: The beat won't show up in some visualizations.

File: vendor/vendor.json (hunk 2 above)
Comment (ruflin) [reply vendor/vendor.json]: we should probably to the same in go-ucfg so we use the exact same version.

**Comment by exekias** [comment]
I just merged #8559, which adds a new use of this lib in `libbeat/management/management.go`, this will need a rebase + update that import :pray:

**Comment by kvch** [comment]
This should be ready for the next round of review.

**Pull request merged**